### PR TITLE
Issue 1013 | Client: Ray provider

### LINKED ISSUE #1013: Client: ray provider
### What is the expected behavior?

Implement ray provider that directly connects to Ray.

### PR DESCRIPTION
**Pull request by IceKhan13**
### Summary

Implementation of Ray provider to connect directly to ray head node

### Details and comments

Closes #1013

### CODE REVIEW

File: client/quantum_serverless/core/job.py [hunk 1]
```diff
@@ -131,21 +132,13 @@ def list(self, **kwargs) -> List["Job"]:
...
-            for key, value in arguments.items():
-                if isinstance(value, dict):
-                    arg_list.append(f"--{key}='{json.dumps(value)}'")
-                else:
-                    arg_list.append(f"--{key}={value}")
-            arguments_string = " ".join(arg_list)
-        entrypoint = f"python {program.entrypoint} {arguments_string}"
+        entrypoint = f"python {program.entrypoint}"
 
         # set program name so OT can use it as parent span name
```
Comment (psschwei) [inline client/quantum_serverless/core/job.py:137]: nit: update comment as we're not just updating program name anymore (?)

File: client/quantum_serverless/core/job.py [hunk 2]
```diff
@@ -131,21 +132,13 @@ def list(self, **kwargs) -> List["Job"]:
...
         env_vars = {
             **(program.env_vars or {}),
             **{OT_PROGRAM_NAME: program.title},
+            **{ENV_JOB_ARGUMENTS: json.dumps(arguments, cls=QiskitObjectsEncoder)},
```
Comment (psschwei) [inline client/quantum_serverless/core/job.py:141]: non-blocking question: is this envvar currently being used? (the github search wasn't super helpful...)

**Review by psschwei — APPROVED** [review]

File: client/quantum_serverless/core/job.py (hunk 2 above)
Comment (IceKhan13) [reply client/quantum_serverless/core/job.py:141]: yes, in `get_arguments` function

File: client/quantum_serverless/core/job.py (hunk 1 above)
Comment (IceKhan13) [reply client/quantum_serverless/core/job.py:137]: what do you mean? :)

Comment (psschwei) [reply client/quantum_serverless/core/job.py:137]: github UI :anger: 

that referred to the comment on L137: it only refers to setting the program name, but now we're also setting job arguments...

File: client/quantum_serverless/core/job.py (hunk 2 above)
Comment (psschwei) [reply client/quantum_serverless/core/job.py:141]: didn't phrase this well, but what I was wondering about if it's being used to set how the program runs ( I probably should've put this on the line where `entrypoint` was defined... ), but looking at it a bit more, it seems like this is just moving the ray client more in line with how the other clients take arguments

Comment (IceKhan13) [reply client/quantum_serverless/core/job.py:141]: exactly :)

File: client/quantum_serverless/core/job.py (hunk 1 above)
Comment (IceKhan13) [reply client/quantum_serverless/core/job.py:137]: ah! :) now I understand. You are right, comment needs to be updated/removed. Will update it in any of next PRs :)

**Pull request merged**